tests/e2e/specs/subscriptions/shopper/shopper-subscriptions-purchase-free-trial.spec.js

```diff
@@ -143,6 +127,8 @@ describeif( RUN_SUBSCRIPTIONS_TESTS )(
 			await expect( page ).toMatchElement( '.trial_end_date', {
 				text: renewalDateFormatted,
 			} );
+
+			await merchant.logout();
```

Comment:
I feel these logouts shouldn't be test specific, they can be set at block level, just review them once to see if we can move them around?

Reply:
I understand that the logout function should be on the teardown block to prevent cascade failures, just in case the test stopped for some reason.

However, what would be the approach where where we use both merchant & shopper logins inside the same test. We can't use `shopper.logout()` or in teardown if the shopper isn't logged in because it would throw an error since selector would be missing.

For `merchant.logout`, it doesn't fail but it won't logout a shopper account.

So I believe we should try and complete the flow. If logged in, we should logout as well. And this would be helpful to troubleshoot tests as well. 

Teardowns are useful when we do repetitive stuff, like a test where only merchant flows are in place.

Let me know your thoughts @deepakpathania

Reply:
I personally feel we should log in and logout in `beforeEach` and `afterEach` blocks for truly independent tests, I understand that's not feasible because some tests in a spec rely on the previous one but we should strive to reach as close to tests being truly isolated from each other as possible. The changes here deviate from that a little as you mentioned above so I feel we should reconsider this.

Reply:
Updated teardowns where possible. Let me know if more changes are required